### PR TITLE
Implement LifecycleExecutor

### PR DESCRIPTION
**Pull request by agunde406**
The LifecycleExecutor will check the database for pending
services that need to be prepared, finalized, retired or purged.

It will check the database on a configured interval or can be
woken up by an alarm. When it finds a pending service, it will
get the commands from the Lifecycle implementation for the service
type and add its own command to update the Lifecycle store. It
then passes all commands to an executor.

The commands should all be very quick so the LifecycleExecutor
only has one thread.

### CODE REVIEW

File: libsplinter/Cargo.toml [hunk 1]
```diff
@@ -129,6 +129,8 @@ experimental = [
     "service-arguments-converter",
     "service-id",
     "service-lifecycle",
+    "service-lifecycle-executor",
```
Comment (peterschwarz) [inline libsplinter/Cargo.toml:132]: This should probably be split into a separate commit

File: libsplinter/src/runtime/service/lifecycle_executor/store/diesel/mod.rs [hunk 2]
```diff
@@ -175,6 +179,90 @@ impl DieselLifecycleStore<diesel::sqlite::SqliteConnection> {
...
+    C: diesel::Connection<TransactionManager = AnsiTransactionManager> + 'static,
+    C::Backend: diesel::backend::UsesAnsiSavepointSyntax,
+{
+    pub fn new(connection: &'a C) -> Self {
+        DieselConnectionLifecycleStore { connection }
+    }
+}
+
+#[cfg(feature = "postgres")]
+impl<'a> LifecycleStore for DieselConnectionLifecycleStore<'a, diesel::pg::PgConnection> {
```
Comment (peterschwarz) [inline libsplinter/src/runtime/service/lifecycle_executor/store/diesel/mod.rs:201]: This seems like it should be in its own commit as well (it's not mentioned in the commit message)

Overall the commit message doesn't cover the purpose of these changes.

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: libsplinter/Cargo.toml (hunk 1 above)
Comment (agunde406) [reply libsplinter/Cargo.toml:132]: split

**Review by peterschwarz — APPROVED** [review]

**Review by isabeltomb — APPROVED** [review]
